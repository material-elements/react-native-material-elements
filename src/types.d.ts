export * from './components/types';
export * from './types/index';
