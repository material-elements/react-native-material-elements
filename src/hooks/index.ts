export * from './useRestyle';
export * from './useThemedProps';
