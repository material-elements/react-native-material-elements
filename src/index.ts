export * from './components';
export * from './hooks';
export * from './libraries';
export * from './utils';
