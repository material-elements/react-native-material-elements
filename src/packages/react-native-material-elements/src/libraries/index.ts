export * from './themes';
export * from './style';
