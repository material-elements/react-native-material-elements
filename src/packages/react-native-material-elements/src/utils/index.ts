export * from './styleGenerator';
export * from './utils';
export * from './dimensions';
