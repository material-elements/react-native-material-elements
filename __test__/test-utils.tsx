import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

export const ThemeWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
