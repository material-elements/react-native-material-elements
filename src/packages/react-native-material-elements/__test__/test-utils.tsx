import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';
import { ThemeProviderProps } from '../src/libraries/types';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  themeProps?: Partial<ThemeProviderProps>;
}

export const ThemeWrapper: React.FC<ThemeProviderProps> = ({ children, ...props }) => {
  return <ThemeProvider {...props}>{children}</ThemeProvider>;
};

export const customRender = (ui: React.ReactElement, options?: CustomRenderOptions) => {
  const { themeProps, ...rest } = options || {};
  return render(ui, {
    wrapper: wrapperProps => <ThemeWrapper {...themeProps}>{wrapperProps.children}</ThemeWrapper>,
    ...rest,
  });
};

export * from '@testing-library/react-native';
export { customRender as render };
