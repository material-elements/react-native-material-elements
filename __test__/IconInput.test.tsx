import React from 'react';
import { render } from './test-utils';
import { IconInput, Text } from '../src';

describe('IconInput', () => {
  const inputTestId = 'icon-input-test-id';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON } = render(<IconInput />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the label of the input', () => {
    const { getByText } = render(<IconInput label="Name" />);
    const label = getByText('Name');
    expect(label).toBeDefined();
  });

  it('should apply the inputWrapperStyles', () => {
    const { getByTestId } = render(<IconInput testID={inputTestId} inputWrapperStyles={{ backgroundColor: 'red' }} />);
    const input = getByTestId(`${inputTestId}-wrapper`);
    expect(input.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });

  it('should render start adornment component', () => {
    const { getByText } = render(<IconInput startAdornment={<Text>start-adornment</Text>} />);
    const adornment = getByText('start-adornment');
    expect(adornment).toBeDefined();
  });

  it('should render end adornment component', () => {
    const { getByText } = render(<IconInput endAdornment={<Text>end-adornment</Text>} />);
    const adornment = getByText('end-adornment');
    expect(adornment).toBeDefined();
  });

  it('should apply start adornment container styles', () => {
    const { getByTestId } = render(
      <IconInput
        testID={inputTestId}
        startAdornment={<Text>start-adornment</Text>}
        startAdornmentContainerStyles={{ backgroundColor: 'red' }}
      />,
    );
    const adornment = getByTestId(`${inputTestId}-start-adornment-container`);
    expect(adornment.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });

  it('should apply end adornment container styles', () => {
    const { getByTestId } = render(
      <IconInput
        testID={inputTestId}
        endAdornment={<Text>end-adornment</Text>}
        endAdornmentContainerStyles={{ backgroundColor: 'red' }}
      />,
    );
    const adornment = getByTestId(`${inputTestId}-end-adornment-container`);
    expect(adornment.props.style).toEqual(expect.objectContaining({ backgroundColor: 'red' }));
  });
});
