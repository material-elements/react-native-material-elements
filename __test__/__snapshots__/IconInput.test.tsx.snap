// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconInput should render correctly 1`] = `
<View>
  <View
    style={
      [
        {
          "backgroundColor": "#ECEFF1",
        },
        {
          "testID": "undefined-wrapper",
        },
        [
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "width": "100%",
          },
          undefined,
          undefined,
        ],
      ]
    }
    testID="undefined-wrapper"
  >
    <TextInput
      placeholderTextColor="#757575"
      style={
        {
          "color": "#212121",
          "flex": 1,
          "minHeight": 48,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
